lib/src/components/Tag/index.tsx

```diff
@@ -0,0 +1,77 @@
+import type { MouseEvent } from 'react'
+import React from 'react'
+
+import { CrossIcon } from '@/components/Icon'
+import { classNames, forwardRefWithAs } from '@/utils'
+
+import tagStyles from './tag.module.scss'
+import type { TagOptions } from './types'
+
+const cx = classNames(tagStyles)
+
+function getTextLength(children: React.ReactNode): number | undefined {
+  if (typeof children === 'string' || typeof children === 'number') {
+    return children.toString().length
+  }
+  if (React.isValidElement(children) && typeof children.props.children === 'string') {
+    return children.props.children.length
+  }
+  return undefined
+}
+
+export const Tag = forwardRefWithAs<TagOptions, 'div'>((props, ref) => {
+  const {
+    as,
+    children,
+    className,
+    icon,
+    onClick,
+    onRemove,
+    removeButtonProps,
+    size = 'md',
+    variant = 'default',
+    ...rest
+  } = props
+
+  const Element = as || 'div'
+
+  // Determine if the tag should have a square shape (only one character and no remove action)
+  const isSquare = getTextLength(children) === 1 && !onRemove
+
+  const handleRemove = (e: MouseEvent<HTMLButtonElement>) => {
+    if (removeButtonProps?.onClick) removeButtonProps.onClick(e)
+    if (onRemove) onRemove()
+  }
+
+  return (
+    <Element
+      className={cx(
+        'root',
+        variant && `variant-${variant}`,
+        size && `size-${size}`,
```

Comment:
You have defaults for variant and size, so no need to check
      `variant-${variant}`,
        `size-${size}`,